go/libraries/doltcore/sqle/stats/write.go

```diff
@@ -139,3 +140,54 @@ func flushStats(ctx *sql.Context, prev prolly.Map, tableStats map[sql.StatQualif
 
 	return m.Map(ctx)
 }
+
+func deleteStats(ctx *sql.Context, prev prolly.Map, quals ...sql.StatQualifier) (prolly.Map, error) {
+	if cnt, err := prev.Count(); err != nil {
+		return prolly.Map{}, err
+	} else if cnt == 0 {
+		return prev, nil
+	}
+
+	sch := schema.StatsTableDoltSchema
+	kd, _ := sch.GetMapDescriptors()
+	var m *prolly.MutableMap
+	m = prev.Mutate()
+	pool := prev.NodeStore().Pool()
+
+	keyBuilder := val.NewTupleBuilder(kd)
+
+	for _, qual := range quals {
+		// delete previous entries for this index
+		keyBuilder.PutString(0, qual.Database)
+		keyBuilder.PutString(1, qual.Table())
+		keyBuilder.PutString(2, qual.Index())
+		keyBuilder.PutInt64(3, 0)
+		firstKey := keyBuilder.Build(pool)
+		keyBuilder.PutString(0, qual.Database)
+		keyBuilder.PutString(1, qual.Table())
+		keyBuilder.PutString(2, qual.Index())
+		keyBuilder.PutInt64(3, 10000)
```

Comment:
Add a comment for which column is being written here, and why it gets a hardcoded 10k.

Reply:
added a `maxBucketFanout` constant to make clearer